### ISSUE TITLE
empty-line-between-blocks should not raise warning when comments exist

### ISSUE THREAD

**Issue by BPScott** [sasstools/sass-lint#228]
Currently comments on the line prior to the first nested selector in a block causes an empty-line-between-blocks warning to be raised. This comment should be treated as a non-existant line and should not raise a warning.

To reproduce, add the following to the bottom of [empty-line-between-blocks.scss](https://github.com/sasstools/sass-lint/blob/develop/tests/sass/empty-line-between-blocks.scss):

```scss
.block {
  // Comment to describe complexity
  .element {
    content: 'baz';
  }
}

.block {
  /* Comment to describe complexity */
  .element {
    content: 'baz';
  }
}
```

I would expect that this would raise no additional linting errors.

---

Interestingly this currently does not occur if the first item within the block (after the comment) is a mixin, extends or media query. The following all pass as expected:

```scss
.block {
  // Comment to describe complexity
  @include foo;
}

.block {
  // Comment to describe complexity
  @extend .foo;
}

.block {
  // Comment to describe complexity
  @media (min-width:50em) {
    content: 'qux';
  }
}
```

**Comment by DanPurdy** [comment]
Yeah, @bgriffith is currently rewriting this rule, he flagged it up when he was working on #216 so would hope to see this resolved shortly.

**Comment by bgriffith** [comment]
Planning on looking at this later tonight. Any thoughts on whether or not the second block should cause a warning as a result of having a declaration? I feel like it should.

```scss
.foo {
  content: 'foo';

  // Foobar
  &__bar {
    content: 'bar';
  }
}

.foo {
  content: 'foo';
  // Foobar
  &__bar {
    content: 'bar';
  }
}
```

**Comment by BPScott** [comment]
Just spotted another edge case for this - Comments at the end of a line throws a warning:

```scss
.item {
  padding-bottom: 10px; // comment

  span {
      padding-bottom: 10px;
  }
}
```

**Comment by bgriffith** [comment]
Good catch @BPScott - will look into it.

**Comment by bgriffith** [comment]
Ok, that's fixed in #258.

**Comment by BPScott** [comment]
:+1:
